src/main/java/com/cityelf/service/UserService.java

```diff
@@ -20,19 +22,23 @@ public UserService() {
     return userRepository.getUsers();
   }
 
-  public User getUser(long id) {
-    return userRepository.getUser(id);
+  public User getUser(long id) throws UserNotFoundException {
+    return userRepository.getUser(id).orElseThrow(() -> new UserNotFoundException());
   }
 
   public void addNewUser(User user) {
     userRepository.addNewUser(user);
   }
 
-  public void updateUser(User user) {
-    userRepository.updateUser(user);
+  public void updateUser(User user) throws UserNotFoundException {
+    if (userRepository.updateUser(user) == false) {
```

Comment:
Do not check condition on **false** or **true**.
You can rewrite it as **if (!userRepository.updateUser(User)) { ... }**